heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraController.java

```diff
@@ -23,9 +23,15 @@
  */
 public interface AuroraController {
 
-  boolean createJob(Map<String, String> auroraProperties);
+  boolean createJob(Map<AuroraField, String> auroraProperties);
   boolean killJob();
-  boolean restartJob(int containerId);
+
+  /**
+   * Restarts a given container, or the entire job if containerId is null
+   * @param containerId ID of container to restart, or entire job if null
+   */
+  boolean restart(Integer containerId);
```

Comment:
when `containerId` is `-1`, the entire job will also be restarted according to [here](https://github.com/twitter/heron/pull/1672/commits/a89f379ddbd6a827ab3dbca014ef8a92c985ee92#diff-c41f12543c6d4d61074654ddf05c9c51R87)